src/utils/maths.ts

```diff
@@ -26,7 +26,8 @@ export function min(array: BigInt[]): BigInt {
 }
 
 export function timesBD(a: BigInt, b: BigDecimal): BigInt {
-  return a.toBigDecimal().times(b).truncate(0).digits;
+  const c = a.toBigDecimal().times(b).truncate(0);
+  return BigInt.fromString(c.toString());
```

Comment:
@JonathanAmenechi Did you see this decimal > string > bigint parsing as necessary?

Reply:
Yes.

without the changes= 20000000000 * 0.5 = 1, which is incorrect.

The changes essentially ensure that the zeroes are present on the BigInt. Without these changes, we were only returning the `digits` variable, which for 20000000000 would be 1.

Making the changes in a local graph yieded the correct results, see the logs below:
```
polymarket-subgraph-graph-node-1  | Mar 29 04:43:27.940 INFO market: 58312306994341893330339332274435578913103689162122171020603488063082490511865 
polymarket-subgraph-graph-node-1  | Mar 29 04:43:27.940 INFO outcomeIndex: 0
polymarket-subgraph-graph-node-1  | Mar 29 04:43:27.940 INFO outcomeTokenPrices[0]: 0.5
polymarket-subgraph-graph-node-1  | Mar 29 04:43:27.940 INFO outcomeTokenPrices[1]: 0.5
polymarket-subgraph-graph-node-1  | Mar 29 04:43:27.940 INFO event.params.amount: 20000000000
polymarket-subgraph-graph-node-1  | Mar 29 04:43:27.940 INFO splitValue: 10000000000 // which is the correct result
```